### PR TITLE
[AIRFLOW-6387] print details of success/skipped task

### PR DESCRIPTION
**Pull request by tooptoop4**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [AIRFLOW-6387](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-6387\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-6387
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-6387\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Love it. This is really useful to know what's going on!

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -963,12 +963,31 @@ def signal_handler(signum, frame):
...
+            try:
+                self.log.info(
+                    'Marking task as SKIPPED.'
+                    'dag_id=%s, task_id=%s, execution_date=%s, start_date=%s, end_date=%s',
+                    self.dag_id,
+                    self.task_id,
+                    self.execution_date.strftime('%Y%m%dT%H%M%S'),
+                    self.start_date.strftime('%Y%m%dT%H%M%S'),
+                    self.end_date.strftime('%Y%m%dT%H%M%S'))
+            except NameError:
```
Comment (feluelle) [inline airflow/models/taskinstance.py]: Is the `try-except` really necessary?

Comment (tooptoop4) [reply airflow/models/taskinstance.py]: wasn't sure where skip exception could be thrown so just a railguard

**Comment by tooptoop4** [comment]
@potiuk pls merge

File: airflow/models/taskinstance.py (hunk 1 above)
Comment (feluelle) [reply airflow/models/taskinstance.py]: No I mean why do you check for `NameError` ?

Comment (tooptoop4) [reply airflow/models/taskinstance.py]: not sure if start_date existed yet

**Comment by tooptoop4** [comment]
@potiuk gentle ping

File: airflow/models/taskinstance.py [hunk 2]
```diff
@@ -964,11 +964,23 @@ def signal_handler(signum, frame):
...
             self.refresh_from_db(lock_for_update=True)
             self.state = State.SKIPPED
+            try:
+                self.log.info(
+                    'Marking task as SKIPPED.'
+                    'dag_id=%s, task_id=%s, execution_date=%s, start_date=%s, end_date=%s',
+                    self.dag_id,
+                    self.task_id,
+                    self.execution_date.strftime('%Y%m%dT%H%M%S') if self.execution_date else '',
+                    self.start_date.strftime('%Y%m%dT%H%M%S') if self.start_date else '',
```
Comment (potiuk) [inline airflow/models/taskinstance.py]:
```suggestion
                    self.start_date.strftime('%Y%m%dT%H%M%S') if hasattr('start_date') and self.start_date else '',
```

File: airflow/models/taskinstance.py [hunk 3]
```diff
@@ -964,11 +964,23 @@ def signal_handler(signum, frame):
...
+                    self.end_date.strftime('%Y%m%dT%H%M%S') if self.end_date else '')
+            except NameError:
```
Comment (potiuk) [inline airflow/models/taskinstance.py]: If you use hasattr, you can remove try/except and you will get log in this case.

**Review by potiuk — COMMENTED** [review]
Just a NIT about NameError handling. 

It would be bad if we will not see this log - people will expect it once they use to it and they will be confused in case they miss it.

File: airflow/models/taskinstance.py (hunk 3 above)
Comment (tooptoop4) [reply airflow/models/taskinstance.py]: fixed

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Cool!

**Comment by tooptoop4** [comment]
@potiuk pls merge

**Comment by potiuk** [comment]
Thanks!

**Pull request merged**